[8.0][website_sale_require_login] Module forcing users to log in to buy

yajo opened the pull request:
Forces the user to log in before buying.

CC @rafaelbn.

yajo committed:
Add module website_sale_require_login.

rafaelbn commented:
Go and buy as you described in readme, after signup you get:

(you must continue with checkout)
- http://3118837-71-2bafa9.runbot1.odoo-community.org/shop/checkout

```
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

```
2015-11-02 18:57:09     WARNING     server  openerp.addons.website.models.ir_http:235 _handle_exception

403 Forbidden:

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/website/models/ir_http.py", line 199, in _handle_exception
    response = super(ir_http, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_http.py", line 145, in _handle_exception
    return request._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/http.py", line 666, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/http.py", line 1370, in get_response
    result.flatten()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/http.py", line 1241, in flatten
    self.response.append(self.render())
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/http.py", line 1235, in render
    context=request.context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/website/models/ir_ui_view.py", line 172, in render
    return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_ui_view.py", line 1017, in render
    return self.pool[engine].render(cr, uid, id_or_xml_id, qcontext, loader=loader, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 262, in render
    return self.render_node(self.get_template(id_or_xml_id, qwebcontext), qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 299, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 446, in render_tag_call
    return self.render(cr, uid, template, d)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 262, in render
    return self.render_node(self.get_template(id_or_xml_id, qwebcontext), qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 301, in render_node
    result = self.render_element(element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 322, in render_element
    g_inner.append(self.render_node(current_node, qwebcontext))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 299, in render_node
    result = self._render_tag[t_render](self, element, template_attributes, generated_attributes, qwebcontext)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 431, in render_tag_if
    if self.eval_bool(template_attributes["if"], qwebcontext):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 237, in eval_bool
    return int(bool(self.eval(expr, qwebcontext)))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 206, in eval
    raise_qweb_exception(message="Could not evaluate expression %r" % expr, expression=expr, template=template)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 203, in eval
    return qwebcontext.safe_eval(expr)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_qweb.py", line 89, in safe_eval
    return eval(expr, None, locals_dict, nocopy=True, locals_builtins=True)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/tools/safe_eval.py", line 314, in safe_eval
    return eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/website_sale_recently_viewed_products/models/website.py", line 11, in recently_viewed_products
    ('sessionid', '=', request.session.sid)], limit=10)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 481, in new_api
    result = method(self._model, cr, uid, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/models.py", line 1645, in search
    return self._search(cr, user, args, offset=offset, limit=limit, order=order, context=context, count=count)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/models.py", line 4666, in _search
    self.check_access_rights(cr, access_rights_uid or user, 'read')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/models.py", line 3504, in check_access_rights
    return self.pool.get('ir.model.access').check(cr, uid, self._name, operation, raise_exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in check
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/tools/cache.py", line 122, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3118837-71-2bafa9/openerp/addons/base/ir/ir_model.py", line 775, in check
    raise openerp.exceptions.AccessError(msg % msg_params)
QWebException: ('AccessError', u'Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Public\n\t- Human Resources/Employee\n\n(Document model: website.sale.product.view)')
```

yajo commented:
README updated. Need to activate the portal.

rafaelbn commented:
I cannot test it, after registering and login it brings you to portal and if you try to go to Website: Internal Server Error

![2015-11-12_4-12-14](https://cloud.githubusercontent.com/assets/8736623/11109698/93656692-88f3-11e5-8060-5b6d99f6b5e1.png)

![2015-11-12_4-07-42](https://cloud.githubusercontent.com/assets/8736623/11109671/43bb1632-88f3-11e5-86a4-477678a8a020.png)

```
2015-11-12 03:03:00     ERROR   server  werkzeug:87 _log

Error on request:
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/sandbox/lib/python2.7/site-packages/werkzeug/serving.py", line 177, in run_wsgi
    execute(self.server.app)
  File "/srv/openerp/instances/openerp-oca-runbot/sandbox/lib/python2.7/site-packages/werkzeug/serving.py", line 165, in execute
    application_iter = app(environ, start_response)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/service/wsgi_server.py", line 216, in application
    return application_unproxied(environ, start_response)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/service/wsgi_server.py", line 202, in application_unproxied
    result = handler(environ, start_response)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 1290, in __call__
    return self.dispatch(environ, start_response)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 1264, in __call__
    return self.app(environ, start_wrapped)
  File "/srv/openerp/instances/openerp-oca-runbot/sandbox/lib/python2.7/site-packages/werkzeug/wsgi.py", line 588, in __call__
    return self.app(environ, start_response)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 1437, in dispatch
    result = ir_http._dispatch()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/website/models/ir_http.py", line 148, in _dispatch
    resp = super(ir_http, self)._dispatch()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_http.py", line 175, in _dispatch
    return self._handle_exception(e)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/website/models/ir_http.py", line 196, in _handle_exception
    return super(ir_http, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_http.py", line 145, in _handle_exception
    return request._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 666, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_http.py", line 171, in _dispatch
    result = request.dispatch()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 684, in dispatch
    r = self._call_function(**self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 310, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 307, in checked_call
    return self.endpoint(*a, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 803, in __call__
    return self.method(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/http.py", line 403, in response_wrap
    response = f(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/web/controllers/main.py", line 477, in web_client
    menu_data = request.registry['ir.ui.menu'].load_menus(request.cr, request.uid, context=request.context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in load_menus

  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/tools/cache.py", line 122, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_ui_menu.py", line 381, in load_menus
    menu_root_ids = self.get_user_roots(cr, uid, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_ui_menu.py", line 355, in get_user_roots
    return self.search(cr, uid, menu_domain, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/mail/mail_group_menu.py", line 42, in search
    context=context, count=False)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_ui_menu.py", line 117, in search
    limit=None, order=order, context=context, count=False)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/models.py", line 1645, in search
    return self._search(cr, user, args, offset=offset, limit=limit, order=order, context=context, count=count)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/models.py", line 4666, in _search
    self.check_access_rights(cr, access_rights_uid or user, 'read')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/models.py", line 3504, in check_access_rights
    return self.pool.get('ir.model.access').check(cr, uid, self._name, operation, raise_exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "<string>", line 2, in check

  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/tools/cache.py", line 122, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3119477-71-75e140/openerp/addons/base/ir/ir_model.py", line 775, in check
    raise openerp.exceptions.AccessError(msg % msg_params)
AccessError: ('AccessError', u'Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Human Resources/Employee\n\t- Administration/Settings\n\n(Document model: ir.ui.menu)')
```

rafaelbn commented:
@Yajo please review travis, thanks

janetrp commented:
internal server error shows trying to sign up
![2015-11-17 17_57_38-500 internal server error](https://cloud.githubusercontent.com/assets/13853118/11218351/e65096c8-8d54-11e5-8b14-93bdb82f42b1.png)

yajo commented:
I found [the guilty line](https://github.com/OCA/e-commerce/commit/daece597f042d64ce59029704d072a58a306ff6e). This should fix many weirdness. Let's see if runbot is happy now.

rafaelbn commented:
**Runbot is happy!**
In checkout I think first button must say Sign in and checkout, the options after ths module is: login or sign up if not you cannot buy.

![18-11-2015 19-34-32](https://cloud.githubusercontent.com/assets/8736623/11250205/9e799318-8e2b-11e5-82be-b59454930458.png)

rafaelbn commented:
After **"sign up and checkout"** I have tried to **sign up** and I got and error as @janetrp said here: https://github.com/OCA/e-commerce/pull/71#issuecomment-157432141

http://3135369-71-daece5.runbot1.odoo-community.org

```
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

yajo commented:
Travis failure is quite tricky. It's a PhantomJS failure, but I think it fails because we change the buy workflow. I cannot disable it in test environment because the change is in a decorator, outside the request. What to do there? Is there a way to overwrite the test?

> In checkout I think first button must say Sign in and checkout

That view is modified by [website_sale_suggest_create_account](https://github.com/OCA/e-commerce/tree/8.0/website_sale_suggest_create_account), which introduces a different workflow for buying. You should have noticed that no matter which button you choose, it leads you to create an account. Having both modules installed makes no sense in a real environment, but in runbot all are installed.

> After "sign up and checkout" I have tried to sign up and I got and error as @janetrp said here: #71 (comment)

I found the guilty module: website_sale_recently_viewed_products. Remove it and it will work. I can fix it if you want.

LeartS commented:
Sorry guys, my fault, the runbot internal server error on signup is because of https://github.com/OCA/e-commerce/pull/79

rafaelbn commented:
Great! Thanks! Ready to review!

rafaelbn commented:
Tested! :+1:  Please before merging change label of button "Checkout without sign up" by "sign in and checkout" as commented here: https://github.com/OCA/e-commerce/pull/71#issuecomment-157813602

rafaelbn commented:
Sorry I mistake, I tested without website_sale_suggest_create_account and it's a BUG. This module must work with out website_sale_suggest_create_account and it doesn't.

It only website_sale_require_login is installed customer must have 2 button:
- Please sign up = "Sign up and checkout"
- Please login = "Sign in and checkout" 

![21-11-2015 18-04-06](https://cloud.githubusercontent.com/assets/8736623/11319560/90bb9292-907b-11e5-93c7-2c54c2f0d2cd.png)

About https://github.com/OCA/e-commerce/pull/71#issuecomment-158098008

a - if you suggest to make an account means:
   a.1 - sing up and checkout
   a.2 - Checkout without sign up (you don't have login and you don't want to have it)
   a.3 - **sign in and chekout (we need 3 buttons)** of course user could have login but he/she is not logged

So module https://github.com/OCA/e-commerce/tree/8.0/website_sale_suggest_create_account is have a **bug that we must fix**. See:

![21-11-2015 18-05-37](https://cloud.githubusercontent.com/assets/8736623/11319541/e338277a-907a-11e5-9135-0fe910deda36.png)

b - if you want to force login means:
   b.1 - you have login --> please log in
   b.2 - you don't have login --> please sign up

![9e799318-8e2b-11e5-82be-b59454930458](https://cloud.githubusercontent.com/assets/8736623/11319604/db305b0e-907c-11e5-940f-5afa22939e31.png)

@pedrobaeza we have here a question: website_sale_suggest_create_account & website_sale_require_login cannot be installed toguether and nobody wants it. Customer will choose option a or option b. I'm thinking that maybe we must create only 1 module with 2 options and this means that this module https://github.com/OCA/e-commerce/tree/8.0/website_sale_suggest_create_account **already merged** must change name or be deprecated.

rafaelbn commented on website_sale_require_login/__openerp__.py:
```diff
@@ -0,0 +1,17 @@
+    "summary": "Force users to login for buying",
+    "version": "8.0.1.0.0",
+    "category": "Website",
+    "website": "https://odoo-community.org/",
```
www.antiun.com

yajo commented:
@rafaelbn You seem confused, but it is not that hard:

The purpose of this module is to **force** the user to log in before buying. Thus, if you click checkout, it leads to login page, and if you do not have an account, you sign up from there too:

![captura del escritorio de 23-11-15 17_14_38](https://cloud.githubusercontent.com/assets/973709/11342211/7afdc47a-9206-11e5-99fd-ae9783bf2661.gif)

Then you have that module `website_sale_suggest_create_account`, which only **suggests** it. That means there is no real logic inside, _just changing the view to make it more clear that we want the user to create an account_, but not forcing it.

Of course, installing both makes no sense. If you _force_ you are not _suggesting_, and vice versa, but in case you find yourself with both installed, then the only problem is that the user will have the _Continue without sign up_ button that is actually asking him to log in. The solution is just to uninstall one of the modules.

The easiest fix is to state in both README files that installing the other module makes no sense.

rafaelbn commented:
OK, then depends on https://github.com/OCA/e-commerce/tree/8.0/website_sale_suggest_create_account to add login button and delete button "Checkout without sign up" will be the best solution

yajo committed:
Improve website_sale_suggest_create_account and depend on it.

Now a new button appears "Log in and checkout".

Module website_sale_require_login depends on this one and removes
the button for checkout without creating an account.

yajo commented:
You are right, it's a better UX. Fixed. This now modifies slightly website_sale_suggest_create_account.

rafaelbn commented:
Tested in runbot, perfect! :+1: 

Only comment an issue of **other module not related with this PR**, please @Yajo  take care of this issue and make a PR to Odoo or OCA (where proceed). I think this must be fixed in https://github.com/OCA/server-tools/tree/8.0/auth_signup_verify_email (by security reasons)

When signing up it email already exits is in the database it show an ugly message like you can see in screenshot.

![2015-11-26_6-34-17](https://cloud.githubusercontent.com/assets/8736623/11416052/6671575a-940a-11e5-8cc9-6116a09d1f10.png)

Please make an adverticing (as you can see in second screen shoot) saying:

Email address already in use:
You indicated you are a new customer, but an account already exists with the e-mail yajo.sk8@gmail.com
Are you a returning customer? -> Sign In
Forgot your password? -> Reset password

So after trying to sign up with an existing e-mail adress we must show links to sign in and reset password.

![2015-11-26_6-59-26](https://cloud.githubusercontent.com/assets/8736623/11416173/603d4244-940b-11e5-91c4-bf6b0138a4dc.png)

cc please @LeartS @dgzurita (maybe @elicoidal you are interested) @sergio-incaser @carlos-incaser @JavierIniesta test this great module! ;-D

elicoidal commented on website_sale_require_login/README.rst:
```diff
@@ -0,0 +1,78 @@
+Require login to checkout
+=========================
+
+This module was written to extend the functionality of eCommerce to force users
```
This module extends the functionality of eCommerce to force users

elicoidal commented on website_sale_require_login/README.rst:
```diff
@@ -0,0 +1,78 @@
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
https

elicoidal commented on website_sale_suggest_create_account/README.rst:
```diff
@@ -7,7 +7,7 @@ Suggest to create user account when buying
 ==========================================
 
 This module was written to extend the functionality of e-Commerce to suggest
```
This module extends the functionality of e-Commerce to suggest

yajo committed:
Fix README.

yajo commented:
@rafaelbn Noted, but let's leave that for another thread.

I'm trying to fix Travis right now.

yajo committed:
Fix PhantomJS test for public user.

yajo commented:
:sweat_smile: That was a hard one, but Travis is happy now. I wish Tours were better documented.

pedrobaeza commented:
Good trick, @Yajo!

:+1:

janetrp commented:
tested in runbot :+1:

rafaelbn commented:
Please @Yajo as commented https://github.com/OCA/e-commerce/pull/71#issuecomment-159814858 open a new thread to fix that. Thanks
